### DISCUSSION TITLE
Word handler, pdf options, and request for more places for documents to be saved to...

### DISCUSSION THREAD

**Discussion by sei-dupdyke** [cmu-sei/GHOSTS#51] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by chipmanfu** [comment]
I have a question regarding the Word handler.  Looking at the source code it seems there is a "pdf-vary-filenames" option that will save the document as a PDF.  However, it also looks like it pulls the save path from CommandArgs[0].  Do I add the "pef-vary-filenames" as a second argument under commandArgs or does it need to be prepended to the save path?   It would be cool if you could put a few save file path's in CommandArgs and then ghost would randomly select one.

**Comment by sei-dupdyke** [comment]
In a word handler config, the commandargs line is just a string array. With all the options enabled, it looks like this:

`"CommandArgs": [ "%homedrive%%homepath%\\Documents", "pdf", "pdf-vary-filenames" ],`

`"%homedrive%%homepath%\\Documents"` == the primary save path for this handler
`"pdf"` == flag for also saving created document as pdf
`"pdf-vary-filenames"` == flag to save the pdf document with a different filename than the word document was saved, leaving this out, they save as exactly the same name, just replacing docx with pdf

For the latter two, they act as flags and so I only need to see if they exist or not. The save path (CommandArgs[0]) could be anything, so I look for it by its position. Over time, the values in CommandArgs have evolved, and maybe this wasn't the best way to architect them! So, I'm thinking of how to add another variable like the one you suggest, that makes more sense in the long term.

Maybe `"save-array:['path1','path2','path3']"` is a better convention going forward? Just thinking out loud here.

I'll look into the suggestion of additional file paths today, that should not be difficult to implement.

**Comment by sei-dupdyke** [comment]
This is implemented and pushed for the current build binaries. The format is:

`"CommandArgs": [ "%homedrive%%homepath%\\Documents", "pdf", "pdf-vary-filenames", "save-array:['c:\\tmp','c:\\tmp\\path2','c:\\tmp\\path3']" ],`

Note the double \\ chars to keep json deserialization happy  :)

**Reply by chipmanfu** [reply]
Thanks for the info!
